### PR TITLE
Wider SRF Head: surface_refine_hidden 192→384 for increased surface prediction capacity

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The SurfaceRefinementHead is the final bottleneck for surface pressure prediction. Currently it has 192 hidden dims and 3 layers (~150K params). The backbone outputs 192-dim hidden states, and the SRF head must compress these into accurate (Ux, Uy, p) predictions at every surface node. If this head is capacity-limited, increasing its width directly improves surface prediction quality.

Evidence that SRF is the bottleneck:
- Adding the SRF head (PR baseline) gave large improvements
- Adding aft-foil SRF (separate SRF for aft-foil nodes) gave additional improvements
- Increasing SRF from 1 to 3 layers gave improvements
- The natural next step: increase width from 192 to 384

This adds ~300K params (trivial vs ~2.7M base) with negligible epoch time increase. VRAM increase is minimal. Zero architecture risk — just a wider MLP.

**Literature:** Width scaling is the simplest capacity increase with the most predictable effect. Wider MLPs improve approximation of complex functions (universal approximation theorem depth).

### Step 1: Change SRF Hidden Dim

Simply change the `--surface_refine_hidden` flag from 192 to 384:

```
--surface_refine_hidden 384
```

Also change the aft-foil SRF to match (if `--aft_foil_srf` is enabled, it should use the same hidden dim by default — check if there's a separate `--aft_foil_srf_hidden` flag and set it to 384 too).

### Step 2: Training Runs

```
cd cfd_tandemfoil && python train.py --agent frieren --seed 42 \
  --wandb_name "frieren/wider-srf-384-s42" \
  --wandb_group "wider-srf-head" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 150 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 384 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature
```

Then seed 73 with same flags but `--seed 73`.

### Important Notes

- This is a ONE-LINE change from baseline: `--surface_refine_hidden 192` → `--surface_refine_hidden 384`
- If there's a separate `--aft_foil_srf_hidden` flag, also set it to 384. If aft-foil SRF inherits from `--surface_refine_hidden`, no extra change needed.
- The wider SRF head has ~4x parameters per layer (384^2 vs 192^2 for hidden→hidden weights). Total SRF param increase: ~450K. This is well within VRAM budget.
- Monitor epoch time — should be nearly identical to baseline since SRF forward pass is fast relative to backbone attention.
- If the wider head overfits (worse OOD metrics), note this in results — it would indicate the SRF is NOT capacity-limited but regularization-limited.

## Baseline

Current best metrics (PR #2251, 2-seed avg):
- p_in: **11.891** (target: < 11.89)
- p_oodc: **7.561** (target: < 7.56)
- p_tan: **28.118** (target: < 28.12)
- p_re: **6.364** (target: < 6.36)
- W&B baseline runs: 7jix2jkg (s42), epkfhxfl (s73)

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT frieren:

## Results

### Metrics (Surface MAE pressure, lower is better)

| Metric | Baseline (2-seed avg) | Wider-SRF s42 | Wider-SRF s73 | Wider-SRF 2-seed avg | Δ vs baseline |
|--------|----------------------|----------------|----------------|---------------------|---------------|
| p_in   | 11.891               | 12.261         | 11.798         | 12.030              | +0.139 (+1.2%) |
| p_oodc | 7.561                | **7.528** ✓    | 7.663          | 7.595               | +0.034 (+0.5%) |
| p_tan  | 28.118               | **27.818** ✓   | 28.668         | 28.243              | +0.125 (+0.4%) |
| p_re   | 6.364                | 6.469          | 6.471          | 6.470               | +0.106 (+1.7%) |

Note: s42 individually beats baseline on p_oodc (-0.4%) and p_tan (-1.1%), but s73 is worse, making the 2-seed average marginally negative.

### Configuration change

- `--surface_refine_hidden 384` (was 192)
- `--aft_foil_srf_hidden 384` (was 192)
- SRF head params: 374,403 (was 113,475) — 3.3x increase
- Aft-foil SRF params: 374,403 (was 113,475) — same 3.3x increase

### Run details

- **W&B run IDs**: `u4g1cf44` (s42), `xcg45eyk` (s73)
- **W&B group**: `wider-srf-head`
- **Epochs**: 148 (s42), 149 (s73) — hit 180-min timeout
- **Best epoch**: 147 (s42), 148 (s73) — both very late, suggesting still improving at cutoff
- **Epoch time**: ~90s/epoch (same as baseline)

### Command used

```bash
cd cfd_tandemfoil && python train.py --agent frieren \
  --wandb_name "frieren/wider-srf-384-s42" \
  --wandb_group "wider-srf-head" \
  --seed 42 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 150 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 384 --surface_refine_layers 3 \
  --aft_foil_srf --aft_foil_srf_hidden 384 --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature
```

### What happened

**Near-neutral result.** The 2-seed average is marginally worse than baseline on all metrics (+0.4-1.7%), but the differences are within the expected run-to-run variance. Seed 42 individually shows improvement on p_oodc (-0.4%) and p_tan (-1.1%), confirming the wider head can find better solutions but fails to do so consistently.

Late best epochs (147/148 out of ~148/149 total) indicate the model was still learning at cutoff — the wider head may simply need more epochs to converge than the 180-min budget allows.

### Analysis

1. **Not capacity-limited, but not harmful either**: The SRF head at 192 hidden dims is NOT the primary bottleneck. The wider head doesn't improve consistently, suggesting the head has sufficient capacity at the current architecture. The backbone's 192-dim hidden states may be the real bottleneck — a wider SRF head can't extract information that isn't in the hidden representations.

2. **Higher complexity + same epoch budget = less converged**: The 3.3x parameter increase requires more gradient steps to converge. With the fixed 180-min budget, the wider model consistently converges later (best epoch 147 vs baseline ~120-130). Given unlimited training, the wider SRF might show improvements.

3. **Seed variance masks real signal**: The difference between s42 (beats on p_oodc, p_tan) and s73 (worse on all) shows the effect size is smaller than the seed-to-seed variance. This means the wider head's impact is not reliably positive at 2 seeds.

### Suggested follow-ups

- **Longer training budget**: If the wider head is worth exploring, a 240-min run (cosine_T_max=200) might allow proper convergence. The late best epochs suggest the 180-min budget is too short for this wider architecture.
- **Wider backbone instead**: Instead of wider SRF, increase n_hidden from 192 to 256. The backbone is more likely the bottleneck since the SRF correction is small-magnitude.
- **SRF only (not aft-foil)**: Try wider SRF head (384) but keep aft-foil SRF at 192. This avoids the 2x parameter increase and focuses capacity on the main foil's surface.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2283 because ADVISOR: Closing — wider SRF head (384) is near-neutral on all metrics (p_in +1.2%, p_oodc +0.5%, p_tan +0.4%, p_re +1.7%). Seed 42 shows individual improvement on p_oodc and p_tan, but seed 73 doesn't — the effect is within run-to-run variance.

**Key insight: SRF head capacity is NOT the bottleneck.** The backbone's 192-dim hidden states limit what the SRF head can extract — a wider head can't surface information that isn't in the representations. The late best epochs (147-148/149) suggest the wider model converges slightly slower but this doesn't translate to better final metrics. Good analysis. Moving you to a fresh direction — input features have the strongest track record..
